python/ray/data/_internal/execution/operators/input_data_buffer.py

```diff
@@ -14,33 +18,89 @@ class InputDataBuffer(PhysicalOperator):
     the arguments for read tasks.
     """
 
-    def __init__(self, input_data: List[RefBundle]):
+    def __init__(
+        self,
+        input_data: Optional[List[RefBundle]] = None,
```

Comment:
We can add an other arg `input_data_factory: Callable[[], List[RefBundle]]` which the planner can use to specify the read task generation logic.

The InputDataBuffer class itself shouldn't know anything about read tasks / readers.

Reply:
@ericl - yeah sure, updated.